src/Elm/Ast.hs

```diff
@@ -24,63 +24,65 @@ import Data.Text (Text)
 
 
 data ElmDefinition
-    = DefAlias ElmAlias
-    | DefType  ElmType
-    | DefPrim  ElmPrim
+    = DefAlias !ElmAlias
+    | DefType  !ElmType
+    | DefPrim  !ElmPrim
     deriving (Show)
 
 data ElmAlias = ElmAlias
-    { elmAliasName   :: Text  -- ^ Name of the alias
-    , elmAliasFields :: NonEmpty ElmRecordField  -- ^ List of fields
+    { elmAliasName      :: !Text  -- ^ Name of the alias
+    , elmAliasFields    :: !(NonEmpty ElmRecordField)  -- ^ List of fields
+    , elmAliasIsNewtype :: !Bool  -- ^ 'True' if Haskell type is a @newtype@
```

Comment:
Isn't this info is applicable from the length of the elmAliasFields?

Reply:
Hmm 🤔 In theory it's possible to write something like this:

```haskell
data Size = Size { unSize :: Int }
```

So this will be `data` instead of `newtype`. So, in theory, it's not enough to check only number of records. In practice, however, I'm not sure that there exist single reason to use `data` and not `newtype` in such cases. Especially when `hlint` complains about such data types as well. So we can not store this value inside AST and write helper functions. There are actually some benefits of not having extra field. Do you think it makes sense to not store it in our AST?

Reply:
You can keep it in type